lib/vdsm/storage/volume.py

```diff
@@ -1381,11 +1381,11 @@ def extendSize(self, new_capacity):
             raise se.VolumeNonWritable(self.volUUID)
 
         volFormat = self.getFormat()
-        if volFormat == sc.COW_FORMAT:
+        if volFormat == sc.COW_FORMAT and self.getType() == sc.SPARSE_VOL:
             self.log.debug("skipping cow size extension for volume %s to "
                            "capacity %s", self.volUUID, new_capacity)
             return
-        elif volFormat != sc.RAW_FORMAT:
+        if volFormat not in [sc.RAW_FORMAT, sc.COW_FORMAT]:
             raise se.IncorrectFormat(self.volUUID)
```

Comment:
I agree, it is not needed. Was not needed before either. With only COW and RAW formats, the elif part was probably never accessed.

I just didn't want to change the logic of the condition in case it was covering a legacy usecase.

Reply:
I agree it is not required to change this in this commit, but it will probably
good idea to make it more clear for the next maintainer. Doing this change in
a separate commit is best.